enhancements/proxy-cache.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Quay as a Proxy Cache
+authors:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+reviewers:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+approvers:
+  - "@fmissi"
+  - "@sleesinc"
+  - "@sdadi"
+creation-date: 2021-12-10
+last-updated: 2021-12-10
+status: implementable
+---
+
+# Quay as a cache proxy for upstream registries
+
+Container development has become widely popular. Customers today rely on container images from upstream registries(like docker, gcp) to get desired services up and running.
+Registries now have rate limitation and throttling on the number of times users can pull from these registries.
+This proposal is to enable Quay to act as a pull through cache where, once pulled images are only pulled again when upstream images have been updated.   
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+ > 1. Will the check against the upstream registry count against an upstream rate limit?
+
+## Summary
+
+Dependencies on container images has increased tremendously with the adoption of container driven development. With the introduction of rate limits
+on popular container registries, Quay will act as a proxy cache to circumvent pull rate limitation from upstream registries. Adding a cache will also
+accelerate pull performance as images are pulled from cache rather than upstream dependencies. Cached images are updated only when the upstream
+image digest differs from cached image. 
+
+### Goals
+
+* A Quay user can define and configure(credentials, staleness period) via config file/app, a repository in Quay that acts a cache for a specific upstream registry.
```

Comment:
Can you describe how the configuration process will look like? Will it be a parameter in the configuration file or will it be configured in the web console (or Quay API)? If the web console, will it be available for all users or only to the superuser?

Reply:
Good point, we need to at least mention how the setup will be addressed here.
From the user stories in the [epic](https://issues.redhat.com/browse/PROJQUAY-465), I understand that regular users should be able to set up proxy orgs:
> * As a Quay user I want to be able to define an organization in Quay that acts a cache for a specific upstream registry
> * As a Quay user I want to be able to supply credentials to the upstream registry when defining a cache organization so that I can circumvent / extend possible pull-rate limits or access private repositories

Reply:
Good point. I will look into this and update once we have a clearer understanding.